### PR TITLE
rpc: Remove duplicate name and argNames from CRPCCommand

### PR DESCRIPTION
**Pull request by maflcko**
Currently, the RPC argument names are specified twice to simplify consistency linting. To avoid having to specify the argnames twice when adding new arguments, remove the linter and add an equivalent test based on RPCHelpMan.

### CODE REVIEW

**Comment by promag** [comment]
Concept ACK

**Comment by stackman27** [comment]
#20012 Code review and Built successful. This definitely looks cleaner than the old version and eliminates lots of repetitions

**Comment by stackman27** [comment]
`#20012` Code review and Built successful. This definitely looks cleaner than the old version and eliminates lots of repetitions

**Comment by stackman27** [comment]
Code review and Built successful. This definitely looks cleaner than the old version and eliminates lots of repetitions

**Comment by MarcoFalke** [comment]
Rebased

**Review by ryanofsky — APPROVED** [review]
Code review ACK fafd58e4dc4089d5aa450feefa9af27f9b34c297. Nice cleanup which gets rid of some parameter repetition dealing with RPCs. It's also nice to drop some c++ parsing code. The previous asserts and new test updates seem to make this pretty foolproof and ensure there's no change in behavior.

**Comment by jonasschnelli** [comment]
Much tidier!
Concept ACK

**Comment by MarcoFalke** [comment]
Rebased, should be trivial to re-ACK

**Comment by MarcoFalke** [comment]
@laanwj As the author of `test/lint/check-rpc-mappings.py`, could you take a look here on the python changes?

**Comment by laanwj** [comment]
ACK fa04f9b4ddffc5ef23c2ee7f3cc72a7c2ae49204
Not sad to see that script go.

File: src/rpc/server.cpp [hunk 1]
```diff
@@ -479,6 +484,18 @@ std::vector<std::string> CRPCTable::listCommands() const
     return commandList;
 }
 
+UniValue CRPCTable::dumpArgMap() const
+{
+    UniValue ret{UniValue::VARR};
+    for (const auto& cmd : mapCommands) {
+        for (const auto& c : cmd.second) {
+            const auto help = RpcMethodFnType(c->unique_id)();
```
Comment (ryanofsky) [inline src/rpc/server.cpp:492]: In commit "rpc: Use RPCHelpMan for check-rpc-mappings linter" (fa92912b4bb4629addcbfdfb7cc000be701614af)

I missed this in initial review, but this cast from unique_id to function pointer isn't really safe. Or at least it results in segfaults with #10102 (https://cirrus-ci.com/task/5469433013469184?command=ci#L2275). The unique_id field was really only supposed to be used to detect RPC aliases, not be converted to a function pointer and called. #21035 updates this to take a different approach making arg retrieval work more like help retrieval.

**Pull request merged**